addons/website_sale/static/src/js/website_sale.js

```diff
@@ -312,8 +312,13 @@ odoo.define('website_sale.website_sale', function (require) {
             for (var k in variant_ids) {
                 if (_.isEmpty(_.difference(variant_ids[k][1], values))) {
                     $.when(base.ready()).then(function() {
+                        if ($price_confirmation.attr("data-product-id") == variant_ids[k][0]) {
```

Comment:
this condition will never be `true`
because variants_ids does not contains the product-id that you are comparing

Take a look this line
https://github.com/odoo/odoo/blob/9.0/addons/website_sale/static/src/js/website_sale.js#L108
And this line 
https://github.com/odoo/odoo/blob/9.0/addons/website_sale/static/src/js/website_sale.js#L124

Variants ids looks like next

![image](https://cloud.githubusercontent.com/assets/10233975/26692008/c84e4ce4-46c4-11e7-8df0-ea1839c2d4f7.png)